app/caldav/auth.py

```
@@ -0,0 +1,58 @@
+            if user.api_key and hmac.compare_digest(str(user.api_key), str(password)):
+                return user.username
+
+            try:
+                if user.check_password(password):
+                    if getattr(user, 'must_change_password', False):
+                        logger.warning(
+                            'CalDAV login refused for %r: password change required',
+                            login)
+                        return ''
+                    return user.username
+            except Exception as exc:  # malformed hash, etc.
```

Ruff (0.16.1)

[warning] Do not catch blind exception: `Exception`

(BLE001)